src/rgw/rgw_admin.cc

```diff
@@ -2319,6 +2315,12 @@ static int bucket_source_sync_status(const DoutPrefixProvider *dpp, rgw::sal::Ra
     return r;
   }
 
+  status.resize(remote_markers.get().size());
```

Comment:
we called `rgw_read_bucket_full_sync_status()` above, so we can use `full_status.shards_done_with_gen.size()` as the local num_shards

we should probably also compare that against `remote_markers.get().size()`, and return EINVAL if the local and remote num_shards don't match

Reply:
both come from remote. we can use any one of them, right?

Reply:
if we call  bucket_sync_status while `shards_done_with_gen` is not fully populated, we will still end up crashing.